WordPress/Classes/Services/AccountService.m

```diff
@@ -260,4 +240,51 @@ - (void)removeWordPressComAccountPasswordIfNeeded
     [defaults synchronize];
 }
 
+- (void)updateUserDetailsForAccount:(WPAccount *)account success:(void (^)())success failure:(void (^)(NSError *error))failure
+{
+    NSAssert(account, @"Account can not be nil");
+    NSAssert(account.username, @"account.username can not be nil");
+    NSAssert(account.xmlrpc, @"account.xmlrpc can not be nil");
+
+    NSString *username = account.username;
+    NSString *xmlrpc = account.xmlrpc;
+    id<AccountServiceRemote> remote = [self remoteForAccount:account];
+    [remote getDetailsForAccount:account success:^(RemoteUser *remoteUser) {
+        // account.objectID can be temporary, so fetch via username/xmlrpc instead.
+        WPAccount *fetchedAccount = [self findAccountWithUsername:username andXmlrpc:xmlrpc];
+        [self updateAccount:fetchedAccount withUserDetails:remoteUser];
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [WPAnalytics refreshMetadata];
+        });
+        if (success) {
+            success();
+        }
+    } failure:^(NSError *error) {
+        DDLogError(@"Failed to fetch user details for account %@.  %@", account, error);
+        if (failure) {
+            failure(error);
+        }
+    }];
+}
+
+- (id<AccountServiceRemote>)remoteForAccount:(WPAccount *)account
+{
+    if (account.restApi) {
+        return [[AccountServiceRemoteREST alloc] initWithApi:account.restApi];
+    }
+    return [[AccountServiceRemoteXMLRPC alloc] initWithApi:account.xmlrpcApi];
+}
+
+- (void)updateAccount:(WPAccount *)account withUserDetails:(RemoteUser *)userDetails
+{
+    account.userID = userDetails.userID;
+    account.username = userDetails.username;
+    account.email = userDetails.email;
+    account.avatarURL = userDetails.avatarURL;
+    if (userDetails.primaryBlogID) {
+        account.defaultBlog = [[account.blogs filteredSetUsingPredicate:[NSPredicate predicateWithFormat:@"blogID = %@", userDetails.primaryBlogID]] anyObject];
+    }
+    [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
```

Comment:
(I thought) the idea was not to call save in the Service themselves, but straight from the caller.
But (i think) i'm wrong, since `saveContext` helper was patched to save anything you throw in (Main or Derived).